### PR TITLE
[Graal.Js] Add Secret Management Functionality to Adaptive Authentication Script

### PR DESCRIPTION
**Pull request by Sachin-Mamoru**
### Proposed changes in this pull request

- Changes are added on top of https://github.com/wso2/carbon-identity-framework/pull/5279
- Introduce Secret Management Functionality: Added the capability to retrieve secrets using the getSecretByName function in the JsGraphBuilder, JsNashornGraphBuilder, and JsOpenJdkNashornGraphBuilder classes.
- Load Secrets Code: Enhanced the FrameworkServiceComponent to load and manage the secrets.js file, providing a JavaScript interface to access secrets.
- Functional Interface for Secret Retrieval: Introduced the GetSecret functional interface for retrieving secrets by name.

Following is the new user experience. Then you can call the secrets.<key> method and assign the secret value to any variable and use it.

image

### Follow up actions

- Add secret management part to the IS and Asgardeo docs.
- Documentation of Best Practices: Best practices should be documented to guide developers in maintaining security.

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java [hunk 1]
```diff
@@ -110,12 +110,14 @@
...
 import org.wso2.carbon.identity.organization.management.service.OrganizationManager;
 import org.wso2.carbon.identity.role.v2.mgt.core.RoleManagementService;
+import org.wso2.carbon.identity.secret.mgt.core.SecretResolveManager;
 import org.wso2.carbon.identity.user.profile.mgt.association.federation.FederatedAssociationManager;
 import org.wso2.carbon.idp.mgt.IdpManager;
 import org.wso2.carbon.idp.mgt.listener.IdentityProviderMgtListener;
 import org.wso2.carbon.stratos.common.listeners.TenantMgtListener;
 import org.wso2.carbon.user.core.service.RealmService;
 
+import javax.servlet.Servlet;
```
Comment (malithie) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java:120]: why this import is needed ?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java [hunk 2]
```diff
@@ -850,6 +854,29 @@ protected void unsetIdentityEventService(IdentityEventService identityEventServi
...
+            name = "secret.config.manager",
+            service = SecretResolveManager.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetSecretConfigManager"
+    )
+    protected void setSecretConfigManager(SecretResolveManager secretConfigManager) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Secret Config Manager is set form functions");
```
Comment (malithie) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java:867]: There's a typo

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/services/ConditionalAuthenticationMgtService.java [hunk 3]
```diff
@@ -44,7 +44,7 @@ public class ConditionalAuthenticationMgtService extends AbstractAdmin {
      */
     public String[] getAllAvailableFunctions() {
 
-        List<String> jsFunctions = Stream.of("executeStep", "selectAcrFrom", "sendError", "Log.info", "require").collect
+        List<String> jsFunctions = Stream.of("executeStep", "selectAcrFrom", "sendError", "Log.info", "require", "secrets.getSecret").collect
```
Comment (malithie) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/services/ConditionalAuthenticationMgtService.java:47]: so you seems to be defining the secret function for graal and nashron.
This is added in the context of nashron only which is quite misleading as this is a separate class that doesn't depends on that JS engine implementation

Comment (Sachin-Mamoru) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/services/ConditionalAuthenticationMgtService.java:47]: resolved with https://github.com/wso2/carbon-identity-framework/pull/5705/commits/900b6de1b94d83e89fd7f62562b6d421f27e94d6

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java (hunk 2 above)
Comment (Sachin-Mamoru) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java]: Resolved with https://github.com/wso2/carbon-identity-framework/pull/5705/commits/1238d7b4e3a5bc7a5e04c595ad86e2bc559c96dd

**Pull request closed without merging**